src/app/route/ProductPage/ProductPage.component.js

```diff
@@ -43,27 +53,37 @@ class ProductPage extends Component {
         this.updateBreadcrumbs();
     }
 
-    /**
-     * Get selected configurable product variant
-     * @param {Object} props
-     * @return {Number} variant index
-     */
-    static getVariantIndexFromProps(props) {
-        const { location: { state: locationState } } = props;
-
-        return (locationState && Object.hasOwnProperty.call(locationState, 'variantIndex'))
-            ? locationState.variantIndex
-            : null;
+    static getDerivedStateFromProps(props, state) {
+        const { id: stateId } = state;
+        const {
+            product: {
+                id,
+                variants,
+                configurable_options
+            },
+            location: { search }
+        } = props;
+
+        if (!(configurable_options && variants && id !== stateId)) return null;
+
+        const parameters = Object.entries(convertQueryStringToKeyValuePairs(search))
+            .reduce((acc, [key, value]) => {
+                if (key in configurable_options) {
+                    return { ...acc, [key]: value };
+                }
+                return acc;
+            }, {});
+
+        if (Object.keys(parameters).length !== Object.keys(configurable_options).length) return { id, parameters };
```

Comment:
For long strings, let's move `return { id, parameters };` to next line in curly brackets :)